cmd/pj-rehearse/main.go

```diff
@@ -114,10 +119,44 @@ func main() {
 		gracefulExit(o.noFail, misconfigurationOutput)
 	}
 
-	jobSpec, err := pjdwapi.ResolveSpecFromEnv()
-	if err != nil {
-		logrus.WithError(err).Error("could not read JOB_SPEC")
-		gracefulExit(o.noFail, misconfigurationOutput)
+	var jobSpec *pjdwapi.JobSpec
+	if o.local {
+		pullRef, err := git(o.candidatePath, []string{"rev-parse", "--abbrev-ref", "HEAD"})
+		if err != nil {
+			logrus.WithError(err).Error("could not get current branch")
+			gracefulExit(o.noFail, misconfigurationOutput)
+		}
+		baseRef, err := git(o.candidatePath, []string{"rev-parse", "--abbrev-ref", pullRef + "@{upstream}"})
+		if err != nil {
+			logrus.WithError(err).Error("current branch has no upstream")
+			gracefulExit(o.noFail, misconfigurationOutput)
+		}
+		baseSHA, err := git(o.candidatePath, []string{"rev-parse", baseRef})
+		if err != nil {
+			logrus.WithError(err).Error("could not parse base revision")
+			gracefulExit(o.noFail, misconfigurationOutput)
+		}
+		pullSHA, err := git(o.candidatePath, []string{"rev-parse", pullRef})
+		if err != nil {
+			logrus.WithError(err).Error("could not parse pull revision")
+			gracefulExit(o.noFail, misconfigurationOutput)
+		}
+		jobSpec = &pjdwapi.JobSpec{
+			Type: pjapi.PresubmitJob,
+			Refs: &pjapi.Refs{
+				Org:     "openshift",
+				Repo:    "release",
+				BaseSHA: baseSHA,
+				BaseRef: baseRef,
+				Pulls:   []pjapi.Pull{{SHA: pullSHA, Ref: pullRef}},
```

Comment:
@stevekuznetsov: is anything horrible going to happen if we don't set the pull request number? We obviously have no value to set it to because these are local changes.

Nothing wrong happened in my tests with a PR number of `0`, but you surely know more of the dark corners of Prow.

Reply:
I think that should be fine

Reply:
This should be empty only in a `periodic` job, right? Since that we rehearse only `presubmits` and also we populate the `jobspec` we won't have any problem with that.

Reply:
This is the fake `JobSpec` that is constructed based on local changes, where we don't have a pull request. The CI code path is unchanged.